README.md

```diff
@@ -1 +1,237 @@
+
+and this to the API Gateway's container definition:
+
+    ports:
+      - 8080:8080
+
+This maps the external port on the host network to the internal port inside the docker-compose network.
+
+If the ports 8080, 5000, or 5001 are already used on the machine you can update the  external port to something like
+
+    - 8090:8080
+
+Inside the container, the it will still listening on port 8080 but everything outside the container can communicate to it using port 8090.
+
+Update the Database API and Data Broker ports for relevant services using the env variables **DATABASE_API_PORT**, **DATA_BROKER_SUB_PORT**, and **DATA_BROKER_PUB_PORT**.
+
+## Volumes
+
+The volumes used by the Gadget are defined at the bottom of the docker-compose file.
+
+    volumes:
+        data-manager-storage:
+        offline-storage:     
+        inline-storage:
+            driver: local
+            driver_opts:
+            o: bind
+            type: none
+            device: /mnt/gadget-inline
+        postgres_db:
+            name: postgres_db
+        static_volume:
+            name: static_volume
+
+It is recommended that the inline-storage volume be bound to a tmpfs folder. Run the following command to create the folder:
+
+    sudo mkdir -p /mnt/gadget-inline && \
+    echo "gadget-inline /mnt/gadget-inline tmpfs rw,nodev,nosuid 0 0" | sudo tee -a /etc/fstab && \
+    sudo mount -a
+
+Otherwise remove the lines from the inline-storage volume definition so it matches offline-storage.
+
+Which services use **offline-storage** and **inline-storage** have already been explained earlier in this document.
+
+**data-manager-storage** is only used by the Data Manager for it's internal book keeping. It should be mounted like this
+
+    volumes:
+      - data-manager-storage:/app/data/internal
+If it is mounted to a different location that will need to be updated using the env variable **INTERNAL_DATA_PATH**
+
+**postgres_db** is only used by the database. It needs to be mounted like this:
+
+    volumes:
+      - postgres_db:/var/lib/postgresql/data
+otherwise the postgres data will not be preserved when the Gadget gets stopped.
+
+**static_volume** is used by the Gadget App and Nginx. It allows Nginx to display the static files in the Gadget App. It needs to be mounted to the Gadget APP like this:
+
+    volumes:
+      - static_volume:/gadgetapp/staticfiles
+
+and to Nginx like this:
+
+    volumes:
+      - static_volume:/gadgetapp/staticfiles
+
+If it is not mounted to that specific location the Gadget App will not work.
+
+## Environment variables
+
+Environment variables can be passed to a container in two ways. They can be defined directly in the docker-compose file like this:
+
+    environment:
+      - ENVIRONMENT_VARIABLE=value
+
+or the container can be given the path to a .env file that where the variables are defined:
+
+    env_file:
+      - ./sensor/sensor.env
+
+## Controlling the Gadget
+
+To build all the custom containers run the command:
```

Comment:
need to add a comma, such as
``To build all the custom containers, run the command:``